shared/code/tce_functions_tcecode.php

```diff
@@ -237,14 +237,14 @@ function F_latex_callback($matches)
             } else {
                 $cmd = 'cd '.K_LATEX_PATH_PICTURE.' && '.K_LATEX_PDFLATEX.' '.$imgpath.'.tex';
                 $sts = exec($cmd, $out, $ret);
-                if (!$sts) {
+                if (!file_exists($imgpath.'.pdf')) {
```

Comment:
Would the following work for you instead?

```suggestion
                if (!$sts || ($ret != 0)) {
```

Reply:
Not working. There is no error in Latex syntax, but all its output in console are displayed. When using file_exists the image is generated and displayed.

Reply:
if $ret=1 there must be something wrong with the conversion.
file_exists has some security issues and I am trying not to use it.
Have you tried testing only for $sts? Would that work for you?